dev/system-test/firestore.ts

```diff
@@ -25,11 +26,21 @@ import {DocumentSnapshot} from '../src/document';
 
 let version = require('../../package.json').version;
 
-type PromiseArgs = {
-  resolve?: Function,
-  reject?: Function,
-  promise: Promise<{}>|null
-};
+class DeferredPromise {
+  resolve: Function;
+  reject: Function;
+  promise: Promise<{}>;
+
+  constructor() {
+    this.resolve = () => {
+      throw new Error('DeferredPromise.resolve has not been initialized');
+    };
+    this.reject = () => {
+      throw new Error('DeferredPromise.reject has not been initialized');
+    };
+    this.promise = Promise.resolve('DeferredPromise.promise has not been initialized');
```

Comment:
Note that I wanted to make this Promise.reject(), but that creates those Node warnings if it is not handled (which it is not).